infra/kdp-core/resources/kdp-cloudtty.cue

```diff
@@ -0,0 +1,447 @@
+				}
+			},
+			{
+				apiVersion: "v1"
+				kind: "Secret"
+				metadata: {
+					name: "general-terminal-token"
+					namespace: "\(parameter.namespace)"
+					annotations: {
+						"kubernetes.io/service-account.name": "general-terminal-sa"
+					}
+				}
+				type: "kubernetes.io/service-account-token"
+			},
+			{
+				kind: "ClusterRole"
+				apiVersion: "rbac.authorization.k8s.io/v1"
+				metadata: {
+					name: "pod-terminal-cr"
+				}
+				rules: [{
+					apiGroups: [""]
+					resources: [
+						"pods",
+						"pods/exec",
+						"pods/log",
+					]
+					verbs: [
+						"get",
+						"list",
+						"exec",
+						"logs",
+						"create"
+					]
+				}]
+			},
+			{
+				kind: "ClusterRole"
+				apiVersion: "rbac.authorization.k8s.io/v1"
+				metadata: {
+					name: "general-terminal-cr"
+				}
+				rules: [
+					{
+						apiGroups: [""]
+						resources: [
+							"pods",
+							"pods/log",
+							"pods/exec"
+						]
+						verbs: [
+							"get",
+							"list",
+							"create",
+							"update",
+							"delete"
+						]
+					},
+					{
+						apiGroups: [""]
+						resources: [
+							"services",
+							"configmaps",
+							"secrets",
+							"persistentvolumes",
+							"persistentvolumeclaims",
+							"endpoints",
+							"events",
+							"limitranges",
+							"resourcequotas",
+							"componentstatuses"
+						]
+						verbs: [
+							"get",
+							"list",
+							"watch",
+							"create",
+							"update",
+							"patch",
+							"delete"
+						]
+					},
+					{
+						apiGroups: ["apps"]
+						resources: [
+							"deployments",
+							"replicasets",
+							"statefulsets",
+							"daemonsets"
+						]
+						verbs: [
+							"get",
+							"list",
+							"watch",
+							"create",
+							"update",
+							"patch",
+							"delete"
+						]
+					},
+					{
+						apiGroups: ["batch"]
+						resources: [
+							"jobs",
+							"cronjobs"
+						]
+						verbs: [
+							"get",
+							"list",
+							"watch",
+							"create",
+							"update",
+							"patch",
+							"delete"
+						]
+					},
+					{
+						apiGroups: [
+							"extensions"
+						]
+						resources: [
+							"ingresses"
+						]
+						verbs: [
+							"get",
+							"list",
+							"watch",
+							"create",
+							"update",
+							"patch",
+							"delete"
+						]
+					},
+					{
+						apiGroups: ["bdc.kdp.io"]
+						resources: [
+							"applications"
+						]
+						verbs: [
+							"get",
+							"list",
+							"create",
+							"update",
+							"delete",
+						]
+					},
+					{
+						apiGroups: [
+							"bdc.kdp.io"
+						]
+						resources: [
+							"bigdataclusters",
+							"contextsecrets",
+							"contextsettings",
+							"xdefinitions",
+						]
+						verbs: [
+							"get",
+							"list"
+						]
+					},
+					{
+						apiGroups: ["cloudshell.cloudtty.io"]
+						resources: [
+							"cloudshells"
+						]
+						verbs: [
+							"get",
+							"list",
+						]
+					},
```

Comment:
Shall we add those resources controlled by big-data components such as flink/kafka/spark operators too?